tests/test_cli_compile.py

```diff
@@ -2457,6 +2447,38 @@ def test_triple_equal_pinned_dependency_is_used(
 )
 
 
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_not_specified_input_file(
+    runner, make_module, fname, content, tmpdir, monkeypatch
+):
+    """
+    It should raise an error if there are no input files or default input files
+    such as "setup.py", "requirements.in" or "pyproject.toml".
+    """
+    make_module(fname=fname, content=content)
+
+    monkeypatch.chdir(tmpdir)
+    out = runner.invoke(cli)
+    monkeypatch.undo()
+
+    assert "small-fake-a==0.1" in out.stderr
+    assert "small-fake-b" not in out.stderr
+    assert "small-fake-c" not in out.stderr
+    assert "extra ==" not in out.stderr
+
+
+def test_not_specified_input_file_without_allowed_files(runner):
+    """
+    It should raise an error if there are no input files or default input files
+    such as "setup.py" or "requirements.in".
+    """
+    out = runner.invoke(cli)
+    assert "If you do not specify an input file" in out.stderr
+    assert out.exit_code == 2
+    for file_name in ("requirements.in", "setup.py", "pyproject.toml"):
+        assert file_name in out.stderr
```

Comment:
```suggestion
    assert out.exit_code == 2
    expected_error = (
        "Error: Invalid value: If you do not specify an input file, the default "
        "is one of: requirements.in, setup.py, pyproject.toml, setup.cfg"
    )
    assert expected_error in out.stderr.splitlines()
```

I'd prefer to check the whole line here

Reply:
Sure, although I'm not a big fan of checking for messages verbatim (for example doesn't work well with i18n). But I can live with that here.